conda-store-server/conda_store_server/build.py

```diff
@@ -416,13 +432,38 @@ def build_constructor_installer(db: Session, conda_store, build: orm.Build):
         conda_store.log, f"creating installer for conda environment={conda_prefix}"
     ):
         with tempfile.TemporaryDirectory() as tmpdir:
+            is_lockfile = build.specification.is_lockfile
+
+            if is_lockfile:
+                specification = schema.LockfileSpecification.parse_obj(
+                    build.specification.spec
+                )
+            else:
+                try:
+                    # Tries to use the lockfile if it's available since it has
+                    # pinned dependencies. This code is wrapped into try/except
+                    # because the lockfile lookup might fail if the file is not
+                    # in external storage or on disk, or if parsing fails
+                    specification = schema.LockfileSpecification.parse_obj(
+                        {
+                            "name": build.specification.name,
+                            "lockfile": json.loads(
+                                conda_store.storage.get(build.conda_lock_key)
+                            ),
+                        }
+                    )
+                    is_lockfile = True
+                except Exception:
```

Comment:
```suggestion
                except Exception as exc:
                    logger.debug("Exception while obtaining lockfile", exc_info=exc)
```

Maybe we should log the exception for easier debuggability down the line.